tsl/src/compression/create.c

```diff
@@ -272,6 +292,7 @@ compresscolinfo_init(CompressColInfo *cc, Oid srctbl_relid, List *segmentby_cols
 							 " timescaledb.compress_segmentby options.")));
 
 		segorder_colindex[col_attno - 1] = i++;
+		colindex[col_attno - 1] = i;
```

Comment:
Same here, shouldn't this be:
```suggestion
		segorder_colindex[col_attno - 1] = i;
		colindex[col_attno - 1] = i++;
```

Reply:
Done.